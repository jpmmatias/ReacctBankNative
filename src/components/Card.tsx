import React from 'react';
import { StyleSheet, View } from 'react-native';
//@ts-ignore
import styled from 'styled-components/native';

const Container = styled.View`
	border-radius: 8px;
	background-color: #fff;
	margin-top: 6px;
	margin-bottom: 6px;
`;

const ContentWrapper = styled.View`
	margin: 27px 20px;
`;
const Card: React.FC = ({ children }) => {
	return (
		<Container
			style={{
				elevation: 3,
				shadowOffset: { width: 1, height: 1 },
				shadowColor: '#333',
				shadowOpacity: 0.3,
				shadowRadius: 2,
			}}
		>
			<ContentWrapper>{children}</ContentWrapper>
		</Container>
	);
};

export default Card;
