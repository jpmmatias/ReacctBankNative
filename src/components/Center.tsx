import React from 'react';
import { View } from 'react-native';
//@ts-ignore
import styled from 'styled-components/native';

interface CenterProps {}

const Container = styled.View`
	flex: 1;
	background-color: #8c52e5;
	align-items: center;
	justify-content: center;
`;

export const Center: React.FC<CenterProps> = ({ children }) => {
	return <Container>{children}</Container>;
};
