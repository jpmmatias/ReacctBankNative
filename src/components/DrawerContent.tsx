import React from 'react';
import { View, StyleSheet, Text, Button } from 'react-native';
import {
	DrawerContentScrollView,
	DrawerItem,
	DrawerContentComponentProps,
} from '@react-navigation/drawer';
import AsyncStorage from '@react-native-async-storage/async-storage';
import { useDispatch } from 'react-redux';
import { Logout } from '../store/modules/user/action';
import { useStore } from 'react-redux';

export function DrawerContent(props: DrawerContentComponentProps) {
	const dispatch = useDispatch();
	const store = useStore();
	return (
		<View
			style={{
				flex: 1,
				justifyContent: 'center',
			}}
		>
			<Text>Dados do Usu√°rioo</Text>
			<Button
				title='Logout'
				onPress={() => {
					AsyncStorage.clear();
					dispatch(Logout());
					console.log(store.getState());
					props.navigation.closeDrawer();
				}}
			/>
		</View>
	);
}
