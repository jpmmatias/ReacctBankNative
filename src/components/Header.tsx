import React from 'react';
import {
	StyleSheet,
	Text,
	View,
	TouchableOpacity,
	Dimensions,
  PixelRatio,
} from 'react-native';
import { FontAwesome5 } from '@expo/vector-icons';
const { width, height } = Dimensions.get('screen');
import { IHeader } from '../types';
const Header = ({ name, onPress }: IHeader) => {
	return (
		<View style={styles.container}>
			<Text style={styles.text}>Ol√° {name}</Text>
			<TouchableOpacity onPress={onPress}>
				<FontAwesome5 name='user-circle' size={24} color='white' />
			</TouchableOpacity>
		</View>
	);
};

export default Header;

const styles = StyleSheet.create({
	container: {
		width: PixelRatio.getPixelSizeForLayoutSize(width * 0.35),
		flexDirection: 'row',
		justifyContent: 'space-between',
		alignItems: 'center',
		marginVertical: PixelRatio.getPixelSizeForLayoutSize((height * 2) / 100),
		paddingHorizontal: (width * 7) / 100,
	},
	text: {
		fontFamily: 'Roboto-Medium',
		fontSize: 22,
		color: '#fff',
	},
});
