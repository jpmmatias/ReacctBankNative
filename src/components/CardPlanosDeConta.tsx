import React from 'react';
import { View, Text, StyleSheet, Dimensions } from 'react-native';
import Card from './Card';
import { FontAwesome5 } from '@expo/vector-icons';
const { width, height } = Dimensions.get('screen');

interface ICardPlanosDeConta {
	receita: string | number;
	despesas: string | number;
}

const CardPlanosDeConta = ({ receita, despesas }: ICardPlanosDeConta) => {
	return (
		<Card>
			<View style={styles.cardTitleWrapper}>
				<FontAwesome5 name='dollar-sign' size={18} color='#9B9B9B' />
				<Text style={styles.cardTitleText}>Planos de conta</Text>
			</View>
			<View>
				<View>
					<Text style={styles.cardBody_planos__text}>
						Tipo do plano: Receita
					</Text>
					<Text style={styles.cardBody_planos_receita__text}>
						R$: {receita}
					</Text>
				</View>
				<View style={styles.cardBody_planos_divisor}></View>
				<View>
					<Text style={styles.cardBody_planos__text}>
						Tipo do plano: Despesas
					</Text>
					<Text style={styles.cardBody_planos_despesas__text}>
						- R$: {despesas}
					</Text>
				</View>
			</View>
		</Card>
	);
};

const styles = StyleSheet.create({
	cardTitleWrapper: {
		flexDirection: 'row',
		alignItems: 'center',
		height: 'auto',
		marginBottom: 14,
	},
	cardTitleText: {
		fontFamily: 'Roboto-Medium',
		color: '#9B9B9B',
		fontSize: 18,
		marginLeft: 10,
	},

	cardBody_planos__text: {
		fontFamily: 'Roboto-Medium',
		fontSize: 15,
		marginBottom: 6,
		color: '#9B9B9B',
	},
	cardBody_planos_receita__text: {
		fontFamily: 'Roboto-Bold',
		fontSize: 28,
		color: '#34A6E7',
	},
	cardBody_planos_divisor: {
		width: (width * 70) / 100,
		height: '0.4px',
		backgroundColor: '#878686',
		marginHorizontal: 'auto',
		marginVertical: 24,
	},
	cardBody_planos_despesas__text: {
		fontFamily: 'Roboto-Bold',
		fontSize: 28,
		color: '#F45F5F',
	},
});

export default CardPlanosDeConta;
