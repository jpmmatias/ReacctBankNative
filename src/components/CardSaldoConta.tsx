import React from 'react';
import { PixelRatio, StyleSheet, Text, View } from 'react-native';
import Card from './Card';
import { FontAwesome5 } from '@expo/vector-icons';

interface ICardSaldoConta {
	saldoDaConta: string | number;
	lancamentoDeDebito: string | number;
}

const CardSaldoConta = ({
	saldoDaConta,
	lancamentoDeDebito,
}: ICardSaldoConta) => {
	return (
		<Card>
			<View style={styles.cardTitleWrapper}>
				<FontAwesome5 name='dollar-sign' size={18} color='#9B9B9B' />
				<Text style={styles.cardTitleText}>Saldo da conta</Text>
			</View>
			<View style={styles.cardBody_saldo}>
				<Text style={styles.cardBody_saldo__text}>R$: {saldoDaConta}</Text>
				<Text style={styles.cardBody_saldo__text_2}>
					Lançamentos de débito: R$ {lancamentoDeDebito}
				</Text>
			</View>
		</Card>
	);
};

export default CardSaldoConta;

const styles = StyleSheet.create({
	cardTitleWrapper: {
		flexDirection: 'row',
		alignItems: 'center',
		height: PixelRatio.getPixelSizeForLayoutSize(15),
		marginBottom: 14,
	},
	cardTitleText: {
		fontFamily: 'Roboto-Medium',
		color: '#9B9B9B',
		fontSize: 18,
		marginLeft: 10,
	},
	cardBody_saldo: {},
	cardBody_saldo__text: {
		color: '#34A6E7',
		fontFamily: 'Roboto-Bold',
		fontSize: 28,
		marginBottom: 10,
	},
	cardBody_saldo__text_2: {
		color: '#9B9B9B',
		fontFamily: 'Roboto-Medium',
		fontSize: 15,
	},
});
