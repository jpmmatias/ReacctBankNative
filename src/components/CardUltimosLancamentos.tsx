import React from 'react';
import { View, Text, StyleSheet, Dimensions } from 'react-native';
import Card from './Card';
import { FontAwesome5 } from '@expo/vector-icons';
import Lancamento from './Lancamento';
const { width, height } = Dimensions.get('screen');

interface ICardUltimosLançamentos {
	lancamentos: any;
}

const CardUltimosLançamentos = ({ lancamentos }: ICardUltimosLançamentos) => {
	return (
		<Card>
			<View style={styles.cardTitleWrapper}>
				<FontAwesome5 name='dollar-sign' size={18} color='#9B9B9B' />
				<Text style={styles.cardTitleText}>Últimos lancamentos</Text>
			</View>
			{lancamentos &&
				lancamentos.map((lancamento: any) => {
					return <Lancamento data={lancamento.data} valor={lancamento.valor} />;
				})}
		</Card>
	);
};

const styles = StyleSheet.create({
	cardTitleWrapper: {
		flexDirection: 'row',
		alignItems: 'center',
		height: 'auto',
		marginBottom: 14,
	},
	cardTitleText: {
		fontFamily: 'Roboto-Medium',
		color: '#9B9B9B',
		fontSize: 18,
		marginLeft: 10,
	},

	cardBody_planos__text: {
		fontFamily: 'Roboto-Medium',
		fontSize: 15,
		marginBottom: 6,
		color: '#9B9B9B',
	},
	cardBody_planos_receita__text: {
		fontFamily: 'Roboto-Bold',
		fontSize: 28,
		color: '#34A6E7',
	},
	cardBody_planos_divisor: {
		width: (width * 70) / 100,
		height: '0.4px',
		backgroundColor: '#878686',
		marginHorizontal: 'auto',
		marginVertical: 24,
	},
	cardBody_planos_despesas__text: {
		fontFamily: 'Roboto-Bold',
		fontSize: 28,
		color: '#F45F5F',
	},
});

export default CardUltimosLançamentos;
