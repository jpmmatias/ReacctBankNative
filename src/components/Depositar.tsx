import React, { useRef, useState } from 'react';
import { View, Text } from 'react-native';
import Card from '../components/Card';
import Button from '../components/Button';
import { AppTabNavProps, IDadosUser, IDepositoConta, ILancamento, IUser } from '../types';
import {
	StyleSheet,
	Image,
	Dimensions,
	TextInput,
	TouchableOpacity,
} from 'react-native';
import { Ionicons } from '@expo/vector-icons';
import { useDispatch, useStore } from 'react-redux';
import api from '../services/api';
import AsyncStorage from '@react-native-async-storage/async-storage';
import { DadosUserInfo } from '../store/modules/user/action';
import Toast from 'react-native-toast-message';
import userReducer from '../store/modules/user/reducer';

const { width, height } = Dimensions.get('window');
const Depositar = ({ navigation, route }: AppTabNavProps<'Depositar'>) => {
	navigation.setOptions({ tabBarVisible: false });
	const inputRef = useRef<TextInput>(null);

	const store = useStore()
	const dispatch = useDispatch()
	const dadosUser:IDadosUser = store.getState().user.dadosUser
	const user:IUser = store.getState().user.user

	const[valor, setValor] = useState(0)
	const[data, setData] = useState('')
	const[descricao, setDescricao] = useState('')


	function fazerDeposito(){
		let deposito:ILancamento = {
			valor:valor,
			data:data,
			descricao:descricao,
			login: user.login,
			conta: dadosUser.contaBanco.id,
			planoConta:838
		};
		
		async function postDeposito( ){
			api
			.post('lancamentos', deposito, {
				headers: {
					'Content-Type': 'application/json',
					Authorization:  await AsyncStorage.getItem('@tokenApp'),
				},
			})
			.then((res) => {
				switch (deposito.planoConta) {
					case 838:
						dadosUser.contaBanco.saldo += deposito.valor;
						dadosUser.contaBanco.lancamentos.push(deposito);
						dispatch(DadosUserInfo(dadosUser));
						break;
					case 840:
						dadosUser.contaBanco.saldo -= deposito.valor;
						dadosUser.contaCredito.saldo += deposito.valor;
						dadosUser.contaCredito.lancamentos.push(deposito);
						dadosUser.contaBanco.lancamentos.push({
							...deposito,
							valor: deposito.valor * -1,
						});
						dispatch(DadosUserInfo(dadosUser));
						break;
					default:
						break;
				}
				Toast.show({
					type: 'success',
					position: 'top',
					text1: 'Oops',
					text2: "Depósito Realizado com Sucesso!",
				});
			})
			.catch((err) => {
				Toast.show({
					type: 'error',
					position: 'top',
					text1: 'Oops',
					text2: err.message,
				});
				dadosUser.contaBanco.saldo += deposito.valor;
				if(dadosUser.contaBanco.lancamentos === undefined){
					dadosUser.contaBanco.lancamentos = [deposito]
				}else{
					dadosUser.contaBanco.lancamentos.push(deposito);
				}
				
				dispatch(DadosUserInfo(dadosUser));
			});
		}

		postDeposito()
		
	}


	return (
		<View style={styles.container}>
			<Image
				style={styles.logo}
				source={require('../assets/images/logo-gamaacademy.png')}
			/>
			<View
				style={{
					width: width,
					paddingLeft: (width * 10) / 100,
					marginBottom: 10,
				}}
			>
				<TouchableOpacity
					onPress={() => {
						navigation.goBack();
					}}
				>
					<Ionicons name='arrow-back' color='#fff' size={22} />
				</TouchableOpacity>
			</View>
			<Card>
				<View style={styles.cardBody}>
					<Image
						style={styles.account}
						source={require('../assets/images/account.png')}
					/>
					<Text style={styles.depos}>Depósitos</Text>
					<View style={styles.form}>
						<TextInput
							value={valor.toString()}
							onChangeText={(value) => value != ''? setValor(parseFloat(value)) : setValor(0)}
							style={[styles.input, { marginBottom: 20 }]}
						></TextInput>
						<TextInput
							placeholder='Data'
							value={data}
							onChangeText={(value) => setData(value)}
							style={[styles.input, { marginBottom: 20 }]}
						></TextInput>
						<TextInput
							placeholder='Descrição'
							value={descricao}
							onChangeText={(value) => setDescricao(value)}
							style={[styles.input, { marginBottom: 50 }]}
						></TextInput>
						<Button
							text='Realizar Depósitos'
							handleClick={fazerDeposito}
							textColor='#fff'
							backgroundColor='#68DE5A'
							textSize={22}
							widthSize={265.64}
							heightSize={56.97}
							textWeight='600'
						/>
					</View>
				</View>
			</Card>
		</View>
	);
};

const styles = StyleSheet.create({
	container: {
		fontFamily: 'Roboto-Regular',
		flex: 1,
		backgroundColor: '#8C52E5',
		alignItems: 'center',
	},
	form: {
		marginBottom: 37,
	},
	logo: {
		height: 55,
		width: 249,
		marginTop: 40,
		marginBottom: 60,
	},
	account: {
		width: 30,
		height: 30,
		marginBottom: -5,
	},
	cardBody: {
		width: (width * 70) / 100,
		height: (height * 60) / 100,
	},
	title: {
		marginTop: 5,
		fontSize: 21,
		textAlign: 'center',
		fontWeight: '500',
		color: '#1d1d1d',
		lineHeight: 24.61,
		marginBottom: 60,
	},
	depos: {
		marginTop: -20,
		textAlign: 'center',
		marginBottom: 60,
		fontSize: 21,
	},
	link: {
		fontWeight: '500',
		fontSize: 13,
		marginBottom: 20,
		color: '#8C52E5',
		textAlign: 'center',
	},
	input: {
		padding: 5,
		borderBottomWidth: 1,
		borderBottomColor: '#878686',
	},
});

export default Depositar;
