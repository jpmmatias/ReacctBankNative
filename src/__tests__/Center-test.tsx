import React from 'react';
import { View } from 'react-native';
import { create } from 'react-test-renderer';
import { Center } from '../components/Center';

const tree = create(
	<Center>
		<View></View>
	</Center>
);

test('center', () => {
	expect(tree).toMatchSnapshot();
});
