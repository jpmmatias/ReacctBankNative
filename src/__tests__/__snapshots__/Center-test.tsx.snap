// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`center 1`] = `
<View
  style={
    Array [
      Object {
        "alignItems": "center",
        "backgroundColor": "#8c52e5",
        "flexBasis": 0,
        "flexGrow": 1,
        "flexShrink": 1,
        "justifyContent": "center",
      },
    ]
  }
>
  <View />
</View>
`;
